src/operators.jl

```diff
@@ -581,16 +581,67 @@ for (dotop,op) in [(:.+,:+), (:.-,:-), (:.*,:*), (:./,:/)]
     @eval begin
         $dotop(lhs::Real,rhs::JuMPTypes) = $op(lhs,rhs)
         $dotop(lhs::JuMPTypes,rhs::Real) = $op(lhs,rhs)
-        $dotop{T<:JuMPTypes}(lhs::Real,rhs::Array{T}) = map(c->$op(lhs,c), rhs)
-        $dotop{T<:JuMPTypes}(lhs::Array{T},rhs::Real) = map(c->$op(c,rhs), lhs)
+        function $dotop{T<:JuMPTypes}(lhs::Real,rhs::Array{T})
+            arr_type = typeof($op(lhs,one(T)))
+            arr = Array(arr_type, size(rhs))
+            @inbounds for i in eachindex(rhs)
+                arr[i] = $op(lhs,rhs[i])
+            end
+            return arr
+        end
+        function $dotop{T<:JuMPTypes}(lhs::Array{T},rhs::Real)
+            arr_type = typeof($op(one(T),rhs))
+            arr = Array(arr_type, size(lhs))
+            @inbounds for i in eachindex(lhs)
+                arr[i] = $op(lhs[i],rhs)
+            end
+            return arr
+        end
         $dotop(lhs::Real,rhs::OneIndexedArray) = $dotop(lhs, rhs.innerArray)
         $dotop(lhs::OneIndexedArray,rhs::Real) = $dotop(lhs.innerArray, rhs)
-        $dotop{T<:JuMPTypes}(lhs::T,rhs::Array) = map(c->$op(lhs,c), rhs)
-        $dotop{T<:JuMPTypes}(lhs::Array,rhs::T) = map(c->$op(c,rhs), lhs)
+        function $dotop{T<:JuMPTypes}(lhs::T,rhs::Array)
+            length(rhs) == 0 && return Array(eltype(rhs),size(rhs))
```

Comment:
That'd look just like this though, returns ArrayAny if its empty but will be tighter if possible. The whole point of doing this is to avoid returning ArrayAnys if possible. I also don't think type stability is going to make difference here.